### PR TITLE
feat(messages): add optional LaTeX math rendering

### LINKED ISSUE #446: Feature Request: Render math formulas in Codex responses (LaTeX support)
### Background
My Codex-assisted projects often involve mathematical derivations and formula-heavy explanations. When CodexMonitor displays Codex responses, math expressions are currently rendered as plain text, which significantly reduces readability and makes it harder to discuss derivations and symbol definitions with the LLM.

### Request
Please add math formula rendering support when parsing/rendering Codex responses, so LaTeX-style math is displayed properly.

### Desired syntax support
- **Inline math**: `$ ... $` and/or `\( ... \)`
- **Block math**: `$$ ... $$` and/or `\[ ... \]`

### Examples (expected to render)
- Inline: `Euler identity: $e^{i\pi}+1=0$`
- Block:
  ```text
  $$
  \nabla \cdot \mathbf{E} = \frac{\rho}{\varepsilon_0}
  $$
  ```

### Implementation suggestions (optional)
- Integrate a math-capable renderer into the Markdown pipeline (e.g., **KaTeX** or **MathJax**).
- Add a setting like `Enable math rendering` (default **off** to avoid breaking changes for existing users).
- Do **not** render math inside fenced code blocks (```), to avoid accidental parsing of code snippets.

### Why this matters
This would greatly improve CodexMonitor’s usability for math-heavy programming work (algorithms, numerical methods, ML, graphics, optimization, etc.), making Codex responses much easier to read and discuss.

Thanks a lot for building and maintaining CodexMonitor—it's been really helpful!

### PR DESCRIPTION
**Pull request by mhulden**
Closes #446

## Summary
Add optional LaTeX math rendering to Codex chat messages.

## What changed
- Added a new setting: **Enable math rendering** (default: off) in Display & Sound > Chat.
- Added markdown math support in chat rendering using:
  - `remark-math`
  - `rehype-katex`
  - `katex`
- Added KaTeX styles to the app bundle.
- Ensured file-link parsing does not interfere with math AST nodes.
- Wired the setting through app state and message rendering paths (local + daemon-backed usage).

## Supported syntax
- Inline: `$...$` and `\(...\)`
- Block: `$$...$$` and `\[...\]`

## Safety behavior
- Math is **not rendered inside fenced code blocks**.

## Validation
- `npm run typecheck`
- `npm run test -- src/features/messages/components/Markdown.test.tsx`
- `npm run test -- src/features/messages/components/Messages.test.tsx`
- `npm run test -- src/features/settings/components/sections/SettingsDisplaySection.test.tsx`
- `cd src-tauri && cargo check`

All passing.

### CODE REVIEW

File: src/features/messages/components/Markdown.tsx [hunk 1]
```diff
@@ -296,6 +299,84 @@ function normalizeListIndentation(value: string) {
...
+    }
+    output.push(normalizeLatexMathDelimitersInChunk(nonFenceChunk.join("\n")));
+    nonFenceChunk = [];
+  };
+
+  for (const line of lines) {
+    if (MARKDOWN_FENCE_PATTERN.test(line)) {
+      flushNonFenceChunk();
+      inFence = !inFence;
+      output.push(line);
```
[the remark this answers is not in the document]
Comment (mhulden) [reply src/features/messages/components/Markdown.tsx]: Good catch — fixed by matching fence marker + minimum length from opener, and added regression test for long-fence/nested-shorter-fence case.

File: src/features/messages/components/Markdown.tsx [hunk 2]
```diff
@@ -296,6 +299,120 @@ function normalizeListIndentation(value: string) {
...
+    (match, body: string) => {
+      const trimmed = body.trim();
+      if (!trimmed) {
+        return match;
+      }
+      return `$$\n${trimmed}\n$$`;
+    },
+  );
+  const withInlineMath = withBlockMath.replace(
+    /\\\(([\s\S]*?)\\\)/g,
```
[the remark this answers is not in the document]
Comment (mhulden) [reply src/features/messages/components/Markdown.tsx]: Thanks. I pushed a follow-up fix that addresses this.

What changed:
- Kept the fence handling fix (marker + minimum length from opener).
- Updated delimiter normalization so `\(...\)` / `\[...\]` conversion no longer mutates markdown link destinations.
- Specifically, link destination segments are masked before delimiter conversion and restored afterward, so URLs like:
  - `[wiki](https://en.wikipedia.org/wiki/Function_\(mathematics\))`
  remain unchanged.
- Fenced and inline code are still protected from delimiter rewriting.

Tests added/updated:
- Existing long-fence regression still passes.
- Added regression test for escaped LaTeX delimiters inside link URLs to ensure no URL rewriting.
- `Markdown.test.tsx` now passes with the new cases, and `npm run typecheck` is clean.

So this should close both concerns:
1. fence marker length correctness
2. no accidental rewriting of non-text markdown syntax (URL destinations).

File: src/features/messages/components/Markdown.tsx [hunk 3]
```diff
@@ -296,6 +299,286 @@ function normalizeListIndentation(value: string) {
...
+        activeFence = null;
+      }
+      continue;
+    }
+
+    const fenceOpener = parseFenceOpener(line);
+    if (fenceOpener) {
+      flushNonFenceChunk();
+      activeFence = fenceOpener;
+      output.push(line);
```
[the remark this answers is not in the document]
Comment (mhulden) [reply src/features/messages/components/Markdown.tsx]: Addressed in `2d48c9e`. `normalizeLatexMathDelimiters` now skips indented code blocks (4-space/tab), so `\\(...\\)` / `\\[...\\]` inside indented code stays literal.

File: src/features/messages/components/Markdown.tsx [hunk 4]
```diff
@@ -296,6 +299,286 @@ function normalizeListIndentation(value: string) {
...
+      }
+      output.push(line);
+      continue;
+    }
+
+    if (BLOCK_LATEX_CLOSE_PATTERN.test(line)) {
+      const body = blockLines.join("\n").trim();
+      if (body) {
+        output.push("$$", body, "$$");
+      } else {
```
[the remark this answers is not in the document]
Comment (mhulden) [reply src/features/messages/components/Markdown.tsx]: Addressed in `2d48c9e`. `\\[...\\]` block conversion now preserves container prefixes (e.g., list/blockquote context), so nested markdown structure is retained.

**Review by xkonjin — COMMENTED** [review]
Automated review, focused on bugs, security, and test coverage:\n\n- Potential secret-handling risk: the diff appears to touch credentials or token-like values. Please verify no secrets are committed, logged, or exposed in client-visible paths.\n\n- Auth-sensitive area: this PR touches authentication/authorization-related code. I\u2019d want explicit tests for privilege boundaries, expired/invalid session handling, and negative authorization cases before merging.\n\n- Test coverage gap: this PR adds non-trivial code, but the diff does not obviously include tests. Please add targeted coverage for the primary success path and at least one failure/edge case.

**Comment by mhulden** [comment]
> Automated review, focused on bugs, security, and test coverage:\n\n- Potential secret-handling risk: the diff appears to touch credentials or token-like values. Please verify no secrets are committed, logged, or exposed in client-visible paths.\n\n- Auth-sensitive area: this PR touches authentication/authorization-related code. I\u2019d want explicit tests for privilege boundaries, expired/invalid session handling, and negative authorization cases before merging.\n\n- Test coverage gap: this PR adds non-trivial code, but the diff does not obviously include tests. Please add targeted coverage for the primary success path and at least one failure/edge case.

1. No auth/session/privilege code paths were touched in this PR update.
2. No secrets or credentials were added/logged/exposed.
3. Added explicit and broad tests for success + edge/failure-style cases in the math normalization path (new scanner test suite + existing markdown/message/settings tests still passing).
